### PR TITLE
[Keyboard] Add LZ erGhost

### PR DESCRIPTION
**Pull request by mechlovin**
## Description
Add LZ erGhost PCB rev. a.
Thank you.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lz/erghost/matrix.c [hunk 1]
```diff
@@ -0,0 +1,442 @@
+/*
```
Comment (zvecr) [inline keyboards/lz/erghost/matrix.c:1]: This needs stripping of all unused code. Keeping the COL2ROW and DIRECT_PINS logic just causes maintenance issues.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/lz/erghost/info.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+    "keyboard_name": "Erghost",
+    "url": "",
+    "maintainer": "Mechlovin' Studio",
+    "width": 18.25,
+    "height": 6.25,
```
Comment (drashna) [inline keyboards/lz/erghost/info.json:6]:
```suggestion
```

File: keyboards/lz/erghost/rules.mk [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/lz/erghost/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/lz/erghost/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (drashna) [inline keyboards/lz/erghost/rules.mk:22]:
```suggestion
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/lz/erghost/rules.mk [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/lz/erghost/rules.mk:15]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/lz/erghost/readme.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# erGhost
+
+![erGhost](https://geekhack.org/index.php?topic=115100.0)
```
Comment (fauxpark) [inline keyboards/lz/erghost/readme.md:3]: This needs to be a direct image link.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**